runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -159,16 +160,31 @@ protected boolean discoverAddressSpace(final UUID streamId,
                 queue.addAll(streamAddressSpace.copyAddressesToSet(maxGlobal));
 
                 long trimMark = streamAddressSpace.getTrimMark();
+
+                // In case we are dealing with a stream that does not have the checkpoint
+                // capability, check to see if we are trying to access an address that has been
+                // previously trimmed.
+                if (!isCheckpointCapable()
+                        && Address.isAddress(trimMark)
+                        && trimMark >= stopAddress) {
+                    String message = String.format("getStreamAddressMap[{%s}] stream has been " +
+                                    "trimmed at address %s and we are trying to access the " +
+                                    "stream starting at address %s.", this, trimMark, stopAddress);
+                    log.info(message);
```

Comment:
But, how will we recover? throwing a TrimmedException, makes upper layers attempt to reset the stream to load from a checkpoint... failing to find one (as the stream is not checkpoint capable)... hence we don't move the pointer and it is always pointing to the last read position. 

Or will this be left to the client to seek up to the new tail?

Reply:
That is correct -- the client has to heal manually by seeking.

Reply:
Just one last thought, we should be OK for now as this stream is not consumed through the object layer. But for nonCheckpointCapable streams accessed through the Object Layer, this TrimmedException would cause a retry->reset->load from checkpoint (which would incur in unnecessary sequencer calls) as there is no checkpoint to load from. Just leaving this note as we might want to improve this later on. For now, it's OK as it will be directly caught by the client.

Reply:
+1 we may need to define a new exception that will bypass the reset and load.